### ISSUE TITLE
Once Proxy setting enabled in dynakube.yaml, it remains in Oneagent config files even removed from yaml/

### ISSUE THREAD

**Issue by melcsanad** [Dynatrace/dynatrace-operator#1862]
**Describe the bug**
DT Operator was deployed with Proxy settings into Openshift Environment. Later it was not required anymore. It was removed (commented and reapplied by dynakube.yaml). The Oneagent pods still keep the proxy config and can't be removed. 

**To Reproduce**
Deploy DT Operator with proxy setting
#dynakube.yaml:
  // Optional: Set custom proxy settings either directly or from a secret with the field 'proxy'
   proxy:
     value: http://proxy.proxy:port

Remove the Proxy setting 
   // Optional: Set custom proxy settings either directly or from a secret with the field 'proxy'
  // proxy:
  //   value: my-proxy-url.com
  //   valueFrom: name-of-my-proxy-secret

Apply the new config yaml.

Oneagent communication still fails because the Oneagent pods keep the Proxy setting but without value:
ruxitagent_host_4064320.0.log:2023-06-02 07:31:08.665 UTC [003e0440] info    [native] Proxy ....................... $(https_proxy)

ruxitagent_host_4064320.0.log:2023-06-05 09:54:58.610 UTC [003e0478] info    [comm  ] URL https://****:9999/communication not working (Unsupported proxy syntax in '$(https_proxy)') (occurred 14 times in the last 1h 0m 0s)

The result is the same if remove the Proxy from CustomResourceDefinition.
It seems these config setting are kept in OA Containers because they survive the delete of dynatrace namspace/CRD too.

**Expected behavior**
By removed proxy settings the DT Operator do not use Proxy anymore.

**Screenshots**
Screenshot of deployment settings after  Proxy was removed: (see $(http_proxy))
![image](https://github.com/Dynatrace/dynatrace-operator/assets/98130057/84149cee-4021-4a30-8400-0a27d1bd954b)

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Chrome
 - Version: 85.0.4183.83

**Additional context**
workaround: Set Proxy attr in Oneagent DaemonSet to empty:
Containers:
  dynatrace-oneagent:
    Container ID:
    Image:         ****:9999/linux/oneagent:latest
    Image ID:
    Port:          <none>
    Host Port:     <none>
    Args:
      --set-host-group=oscp4-prod_r-it
      --set-network-zone=oscp4-prod_r-it
      --set-proxy=
In this case the proxy setting will be orverwritten and clean up the $(http_proxy) envvar from config files.

**Comment by wepudt** [comment]
Hi
just tested it but could not reproduce the described behaviour. Perhaps the problem occurs when setting an invalid proxy- I tried with a working nginx one.

As a result we've identified and tagged the issue as a "Support request".

Dynatrace responds to requests like these via [Dynatrace ONE support](https://support.dynatrace.com/) rather than Github. This helps our team respond as quickly as possible using the support team's tools and procedures.

Thanks for your help!
